File: 1.4/4.Queue/Queue/queue.py
class Queue:
    def __init__(self):
        self.queue = []
        # инициализация хранилища данных

    def enqueue(self, item):
        self.queue.append(item)
        # вставка в хвост

    def dequeue(self):
        # выдача из головы
        if self.size() > 0:
            return self.queue.pop(0)
        else:
            return None # если очередь пустая

    def size(self):
        return len(self.queue)

    def rotateQueue(self, N: int):
        for i in range(N):
            self.enqueue(self.dequeue())
            i += 1

File: 2.b.1/Queue/Queue.cs
abstract class Queue<T>
{
    // Конструктор
    public Queue<T> Queue(); // постусловие: создана новая пустая очередь

    // команды:
    public void enqueue(T value); // постусловие: в конец очереди добавлен новый элемент

    // предусловие: очередь не пустая
    public T dequeue(); // постусловие: из очереди удалён и возвращён элемент

    // запросы:
    public int size(); // постусловие: размер очереди возвращён

    // предусловие: очередь не пустая
    public T get(); // постусловие: первый элемент очереди возвращён

    // запросы статусов (возможные значения статусов)
    public int get_dequeue_status(); // успешно; очередь пустая
    public int get_get_status(); // успешно; очередь пустая
}